localstack/services/cloudformation/models/ec2.py

```diff
@@ -315,6 +319,25 @@ def fetch_state(self, stack_name, resources):
 
     @staticmethod
     def get_deploy_templates():
+        def _pre_delete(resource_id, resources, resource_type, func, stack_name):
+            res = resources[resource_id]
+            state = res[
+                KEY_RESOURCE_STATE
+            ]  # TODO: there is probably a better way to get the state here
+            physical_resource_id = res["PhysicalResourceId"] or state.get("VpcId")
```

Comment:
nit: Can we assume that `PhysicalResourceId` is always set here, or perhaps better:

```
physical_resource_id = res.get("PhysicalResourceId") or state.get("VpcId")
```

In fact, once we've obtained a typed instance above, we can also use the `physical_resource_id` property:
```
physical_resource_id = resource.physical_resource_id or state.get("VpcId")
```

Reply:
Not sure why this was actually in there since it isn't really being used. I've removed this part